### PR TITLE
Add execution support for initEnv

### PR DESCRIPTION
**Pull request by dodie**
From now initEnv can run commands in the bootstrapped environment.

### CODE REVIEW

File: src/main/resources/initEnv [hunk 1]
```diff
@@ -46,3 +46,11 @@ fi
 set -a
 eval "$VARS"
 set +a
+
+not_sourced() {
+	test "X$(basename -- "$0")" = "XinitEnv"
+}
+if not_sourced; then
+	# when the script is invoked directly with parameters execute them in the bootstrapped context
+	$@
```
Comment (sandro-h) [inline src/main/resources/initEnv]: Shouldn't this rather be `"$@"` to preserve quoted arguments?
E.g. if I want to run `./initEnv npm test --myarg="hello world"` it will pass `--myarg=hello` and a separate `world` arg to `npm test`

I always find this topic a bit hazy. For example I don't know if there are opposite edge cases where `$@` without quotes is necessary.

Comment (dodie) [reply src/main/resources/initEnv]: You are right, without the quotes it will behave just like you described, and it's not good. I'll update the PR.

**Review by ghost — APPROVED** [review]

**Review by sandro-h — APPROVED** [review]

**Review by nemeth-z — APPROVED** [review]

File: src/main/resources/initEnv [hunk 2]
```diff
@@ -46,3 +46,11 @@ fi
 set -a
 eval "$VARS"
 set +a
+
+not_sourced() {
+	test "X$(basename -- "$0")" = "XinitEnv"
```
Comment (mcico) [inline src/main/resources/initEnv:51]: **question**: What is the capital X for?

**Review by mcico — CHANGES_REQUESTED** [review]

File: src/main/resources/initEnv (hunk 2 above)
Comment (ghost) [reply src/main/resources/initEnv:51]: A proof of how ugly shell scripts can get when needed to be written for the greatest compatibility ever. :)
https://stackoverflow.com/questions/19783948/portable-way-to-check-emptyness-of-a-shell-variable

**Edit:** Found a better link:
https://stackoverflow.com/questions/174119/why-do-shell-script-comparisons-often-use-xvar-xyes

**Review by hosszubalazs — APPROVED** [review]

File: src/main/resources/initEnv (hunk 2 above)
Comment (dodie) [reply src/main/resources/initEnv:51]: Context: This function was introduced in an old version of initEnv, but it was removed in my previous PR.
With the new feature added in this PR, the function was needed again, so I put it back.

I think in most cases the X padding is not needed, and I was not able to introduce any issues by removing it. However, I found some old forum threads recommending this format. Because initEnv is expected to run on many environments, my guess is that it was introduced to overcome a quirk of `test` in certain cases.

Comment (mcico) [reply src/main/resources/initEnv:51]: Thanks for the explanation.

**Review by mcico — APPROVED** [review]
